mautic/mautic
# Site Url is validated in the UI to be a valid url

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [:heavy_multiplication_x:]
| New feature/enhancement? (use the a.x branch)      | [:heavy_check_mark:]
| Deprecations?                          | [:heavy_multiplication_x: ]
| BC breaks? (use the c.x branch)        | [:heavy_multiplication_x:]
| Automated tests included?              | [:heavy_multiplication_x:] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


This PR prevents saving the url field in the config form without a protocol. If the url value does not start with any protocol (e.g. https://) when submitting the configuration form, the `https` protocol will be prepended to the string.

![image](https://github.com/mautic/mautic/assets/96085911/a011ba0a-f0b1-4d85-9a0c-ca837972ade2)
is saved as
![image](https://github.com/mautic/mautic/assets/96085911/bebd645c-889c-4fad-9d34-7e2d71286f07)

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open the configuration page.
3. Open System Settings and try to save the `Site URL` field without a protocol. E.g. `mautic.ddev.site`.
4. `https://mautic.ddev.site` is saved.
